logservice/logpuller/region_request_worker.go

```diff
@@ -379,8 +380,7 @@ func (s *regionRequestWorker) processRegionSendTask(
 			for _, state := range s.takeRegionStates(subID) {
 				state.markStopped(&requestCancelledErr{})
 				regionEvent := regionEvent{
-					state:  state,
-					worker: s,
+					states: []*regionFeedState{state},
```

Comment:
add one new method named `newRegionFeedState` and check the passed in state has at least one element, otherwise just panic.